src/App.tsx

```diff
@@ -1,26 +1,134 @@
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    loadTodos();
+  }, []);
+  useEffect(() => {
+    if (errorMessege === null) {
+      return;
+    }
+
+    const timer = setTimeout(() => setErrorMessege(null), 3000);
+
+    return () => clearTimeout(timer);
+  }, [errorMessege]);
+
+  async function handleAddTodo(event: React.FormEvent) {
+    event.preventDefault();
+    if (!newTodo.trim()) {
+      setErrorMessege('Title should not be empty');
+
+      return;
+    }
+
+    try {
+      setLoading(false);
+      const createdTodo = await addTodos(newTodo);
+
+      setTodos([...todos, createdTodo]);
+      setNewTodo('');
+    } catch {
+      setLoading(true);
+      setErrorMessege('Unable to add a todo');
+    } finally {
+      setLoading(false);
+      setErrorMessege(null);
+    }
+  }
+
+  async function handleDeleteTodo(id: number) {
+    try {
+      setLoading(false);
+      deleteTodos(id);
+      setTodos(await getTodos());
+    } catch {
+      setLoading(true);
+      setErrorMessege('Unable to delete a todo');
+    } finally {
+      setLoading(false);
+    }
+  }
+
+  const filteredTodos = () => {
+    switch (selectedFilter) {
+      case Filter.Active:
+        return todos.filter(todo => !todo.completed);
+      case Filter.Completed:
+        return todos.filter(todo => todo.completed);
+      case Filter.All:
+        return todos;
+      default:
+        return todos;
+    }
+  };
+
+  const handleClearCompleted = () => {
+    setTodos(todos.filter(todo => !todo.completed));
+  };
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">
-          React Todo App - Load Todos
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+      <div className="todoapp__content">
+        <Header
+          todos={todos}
+          newTodo={newTodo}
+          setNewTodo={setNewTodo}
+          loading={loading}
+          handleAddTodo={handleAddTodo}
+          loadTodos={loadTodos}
+        />
+        <Section todos={filteredTodos()} handleDeleteTodo={handleDeleteTodo} />
+        {/*+ Hide the footer if there are no todos */}
+        {todos.length > 0 && (
+          <Footer
+            todos={todos}
+            selectedFilter={selectedFilter}
+            setselectedFilter={setselectedFilter}
+            handleClearCompleted={handleClearCompleted}
+          />
+        )}
+      </div>
+      {loading && <Loader />}
+
+      {(errorMessege === null || todos.length === 0) && (
+        <Error errorMessege={errorMessege} setError={setErrorMessege} />
+      )}
```

Comment:
Error component should be rendered always, just hide it with toggling hidden class